lib/libutils/ext/include/compiler.h

```diff
@@ -246,4 +249,25 @@
 #define __compiler_atomic_store(p, val) \
 	__atomic_store_n((p), (val), __ATOMIC_RELAXED)
 
+#ifdef __HAVE_BUILTIN_SPECULATION_SAFE_VALUE
+#define __compiler_speculation_safe_value(v) __builtin_speculation_safe_value(v)
```

Comment:
How does this relate to `__builtin_load_no_speculate()`? (see https://github.com/OP-TEE/optee_os/blob/master/lib/libutils/ext/include/speculation_barrier.h#L30-L45).

If it is a more "modern" replacement, then I think `speculation_barrier.h` should be removed (it is not used anymore since ldelf was introduced anyways).

Reply:
I think `__builtin_load_no_speculate()` was/is an arm extension to GCC. I don't know the status.
However, `__builtin_speculation_safe_value()` is used in Trusted Firmware A so that says a bit.